src/tasks.ts

```diff
@@ -1,87 +1,140 @@
+        'Build',
+        {
+            type: TYPE,
+            code: [ 'devtools::build()' ]
+        },
+        vscode.TaskGroup.Build
+    ),
+
+    new TaskMeta(
+        'Check',
+        {
+            type: TYPE,
+            code: [ 'devtools::check()' ]
+        },
+        vscode.TaskGroup.Test
+    ),
+    
+    new TaskMeta(
+        'Document',
+        {
+            type: TYPE,
+            code: [ 'devtools::document()' ]
+        },
+        vscode.TaskGroup.Build
+    ),
+    
+    new TaskMeta(
+        'Install',
+        {
+            type: TYPE,
+            code: [ 'devtools::install()' ]
+        },
+        vscode.TaskGroup.Build
+    )
+];
+
+
+const asTask = function (
+    folder: vscode.WorkspaceFolder | vscode.TaskScope,
+    meta: TaskMeta
+): vscode.Task {
+    const task: vscode.Task = new vscode.Task(
+        meta.definition,
+        folder,
+        meta.name,
+        TYPE,
+        new vscode.ProcessExecution(
+            'Rscript',
+            ['-e', meta.definition.code.join('; ')],
+            {
+                cwd: meta.definition.cwd,
+                env: meta.definition.env
+            }
+        ),
+        meta.problemMatchers
+    );
+    
+    if (meta.group) {
+        task.group = meta.group;
     }
+    
+    return task;
+};
+
 
-    private readonly tasks = [
-        this.getTask('Build', {
-            type: this.type,
-            command: 'Rscript',
-            args: [
-                '-e',
-                'devtools::build()'
-            ]
-        }),
-        this.getTask('Check', {
-            type: this.type,
-            command: 'Rscript',
-            args: [
-                '-e',
-                'devtools::check()'
-            ]
-        }),
-        this.getTask('Document', {
-            type: this.type,
-            command: 'Rscript',
-            args: [
-                '-e',
-                'devtools::document()'
-            ]
-        }),
-        this.getTask('Install', {
-            type: this.type,
-            command: 'Rscript',
-            args: [
-                '-e',
-                'devtools::install()'
-            ]
-        }),
-        this.getTask('Test', {
-            type: this.type,
-            command: 'Rscript',
-            args: [
-                '-e',
-                'devtools::test()'
-            ]
-        }, '$testthat'),
-    ];
 
-    public provideTasks(): vscode.Task[] {
-        return this.tasks;
+export class RTaskProvider implements vscode.TaskProvider {
+    
+    public type = TYPE;
+
+    public provideTasks(): vscode.Task[] {        
+        const folders = vscode.workspace.workspaceFolders;
+        
+        if (!folders) {
+            return [] as vscode.Task[];
+        }
+        
+        const tasks: vscode.Task[] = [];
+        
+        for (const folder of folders) {
+            const is_r_workspace = fs.existsSync(path.join(folder.uri.fsPath, 'DESCRIPTION'));
+            if (is_r_workspace) {
```

Comment:
`DESCRIPTION` is used to identify workspaces that contain an R package. Should it be named `isRPackage`?

Reply:
Changed